### PR TITLE
Add board testing

### PR DESCRIPTION
**Pull request by stelznertim**
all test cases and according methods for the board class

### CODE REVIEW

File: Connect-Four/spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (svs-invision) [inline Connect-Four/spec/main_spec.rb:1]: you commit an empty file?

File: Connect-Four/spec/player_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      expect(player.name).to eq('Tim')
+    end
+
+    it 'has right symbol for disc' do
+      expect(player.disc).to eq('▵')
+    end
+  end
+
+  context 'let player select a column for his disc' do
+    it 'selects disc' do
```
[the remark this answers is not in the document]
Comment (stelznertim) [reply Connect-Four/spec/player_spec.rb:22]: --> test case 'let player select a column for his disc' not needed, will delete it

File: Connect-Four/spec/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+require_relative '../lib/board'
+
+describe Board do
+  let(:board) { Board.new }
+  context 'check that board is initialized correctly' do
```
Comment (stelznertim) [inline Connect-Four/spec/board_spec.rb:5]:
```suggestion
  context 'check that board is initialized correctly' do
  board = Board.new
```

File: Connect-Four/spec/board_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    it 'initialized rows correctly' do
+      expect(board.rows).to eq(6)
+    end
+    it 'intializes columns correctly' do
+      expect(board.columns).to eq(7)
+    end
+    it 'initializes the grid correctly' do
```
Comment (stelznertim) [inline Connect-Four/spec/board_spec.rb:12]: leave out last test

File: Connect-Four/spec/board_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      expect(board.grid.flatten.size).to eq(42)
+    end
+  end
+
+  context 'check for valid_turn? ' do
```
Comment (stelznertim) [inline Connect-Four/spec/board_spec.rb:17]: context 'turn is valid'
  it 'player has entered 5' do

File: Connect-Four/spec/board_spec.rb [hunk 6]
```diff
@@ -2,46 +2,100 @@
...
+  end
+
+  context 'the initialized attributes of board are wrong' do
+    it ' the number of rows is not 7' do
+      expect(board.rows).not_to eq(7)
+    end
+    it 'the number of columns is not 8' do
+      expect(board.columns).not_to eq(8)
+    end
+  end
```
Comment (stelznertim) [inline Connect-Four/spec/board_spec.rb:22]: not needed

File: Connect-Four/spec/board_spec.rb [hunk 7]
```diff
@@ -2,46 +2,100 @@
...
-      expect(board.valid_turn?(column)).to be false
+  end
+  context ' the column is full' do
+    before do
+      column = 4
+      6.times do
+        board.update(column, '©')
+      end
+    end
+    it 'a full column 4 is not valid' do
```
Comment (stelznertim) [inline Connect-Four/spec/board_spec.rb:42]: throw exception as error feedback

File: Connect-Four/lib/board.rb [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  end
+
+  def check_rows(input)
+    grid.each do |row|
+      return true if row[input - 1].nil?
+    end
+    false
+  end
+
+  def update(column, symbol)
```
Comment (stelznertim) [inline Connect-Four/lib/board.rb:21]: double validation needed to intercept mutating after the frontend validation

**Pull request merged**